src/cljs/org/broadinstitute/firecloud_ui/page/method_repo.cljs

```diff
@@ -49,12 +53,21 @@
 
 (react/defc Page
   {:render
-   (fn [{:keys [state]}]
+   (fn [{:keys [state refs]}]
      [:div {:style {:padding "1em"}}
       [:h2 {} "Method Repository"]
       [:div {}
        (cond
-         (:methods-loaded? @state) [MethodsList {:methods (:methods @state)}]
+         (:methods-loaded? @state) [:div {}
+                                    (let [do-filter #(swap! state assoc :filtered-methods
```

Comment:
I suggest a slight restructuring here. Define a function `filter-methods` that takes a list of methods and a filter value and returns the filtered list. Then, you would create the `MethodsList` component like
```
[MethodsList {:methods (filter-methods (:methods @state) (:filter-text @state))}]
```
In the places you call `do-filter`, instead do `(swap! state assoc :filter-text (.-value (.getDOMNode (@refs "input"))))`